From: Yury Norov <yury.norov@gmail.com>
Subject: lib/vsprintf: don't increment buf in bitmap_list_string

Increment is confusing as the buf is overritten at the same line.

Link: https://lkml.kernel.org/r/20210817193735.269942-1-yury.norov@gmail.com
Fixes: b1c4af4d3d6b (vsprintf: rework bitmap_list_string) (next-20210817)
Signed-off-by: Yury Norov <yury.norov@gmail.com>
Suggested-by: Andy Shevchenko <andy.shevchenko@gmail.com>
Cc: Wolfram Sang <wsa+renesas@sang-engineering.com>
Cc: Alexander Lobakin <alobakin@pm.me>
Cc: Alexey Klimov <aklimov@redhat.com>
Cc: Dennis Zhou <dennis@kernel.org>
Cc: Jiri Olsa <jolsa@redhat.com>
Cc: Ulf Hansson <ulf.hansson@linaro.org>
Cc: Will Deacon <will@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/vsprintf.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/lib/vsprintf.c~vsprintf-rework-bitmap_list_string-fix
+++ a/lib/vsprintf.c
@@ -1261,7 +1261,7 @@ char *bitmap_list_string(char *buf, char
 
 		if (buf < end)
 			*buf = '-';
-		buf = number(++buf, end, rtop - 1, default_dec_spec);
+		buf = number(buf + 1, end, rtop - 1, default_dec_spec);
 	}
 	return buf;
 }
_
